.github/workflows/build_conda_linux.yml

```
@@ -99,20 +111,18 @@ jobs:
         run: |
           set -euxo pipefail
           cat "${{ inputs.env-var-script }}" >> "${BUILD_ENV_FILE}"
+      - uses: ./test-infra/.github/actions/cache-path
+        if: ${{ inputs.cache-path != '' }}
+        with:
+          cache-path: ${{ inputs.cache-path }}
+          cache-key: ${{ inputs.cache-key }}
+          repository: ${{ inputs.repository  }}
+          cache-miss-script: ${{ inputs.cache-miss-script }}
```

I'm thinking the following semantics for using this job might be cleaner:

```suggestion
      - name: Run Pre-Script with Caching
        uses: ./test-infra/.github/actions/cache-path
        with:
          cache-path: ${{ inputs.cache-path }}
          cache-key: ${{ inputs.cache-key }}
          repository: ${{ inputs.repository  }}
          cache-miss-script: ${{ inputs.pre-script }}
```

And then we can remove the next step entirely. This kind of uses the `cache-path` action as a generic script-runner with optional caching. If `cache-path` is not defined, then it is just a vanilla script-runner, so there is no need for a separate `run-script` action. And we can probably rename it/update the description to reflect this usage better.

I don't think this build workflow needs a `cache-miss-script`  input here. For example, say we want to cache the env for the post-script in the future as well - then we don't want to require uses to provide `cache-miss-script` inputs for each place this workflow uses caching.

---

> Hmm, we already have logic for `conda` environment caching in core, can it perhaps be reused here.
> 
> If it's a more generic caching mechanism, it would have benefited from more detailed description in PR how this caching is achieved.
> 
> cc: @huydhn

Exactly same caching mechanism is used in conda caching here: https://github.com/pytorch/test-infra/blob/main/.github/actions/setup-miniconda/action.yml#L40

---

Yup, this is the same GitHub cache action used in setting up conda. As a generic action, the logic looks fine IMO.  One comment is that I think we can combine `run-script` and `run-script-with-cache` into one and just call it `run-script`.  The reason is that the cache would only be used when `cache-path` is not empty.  And when it's empty, `run-script-with-cache` and `run-script` behave exactly the same

---

Really great idea! Thank you. Done.